misc/peer-store/src/behaviour.rs

```diff
@@ -0,0 +1,179 @@
+use std::{collections::VecDeque, task::Poll};
+
+use libp2p_core::{Multiaddr, PeerId};
+use libp2p_swarm::{dummy, NetworkBehaviour};
+
+use crate::store::{AddressSource, Store};
+
+/// Events generated by [`Behaviour`] and emitted back to the [`libp2p_swarm::Swarm`].
+pub enum Event<S: Store> {
+    /// The peer's record has been updated.  
+    /// Manually updating a record will always emit this event
+    /// even if it provides no new information.
+    RecordUpdated {
+        peer: PeerId,
+    },
+    Store(S::ToSwarm),
+}
+
```

Comment:
`Behaviour`, and therefore `Event`, needs to impl `Debug` for the `NetworkBehaviour` derive macro to work. Putting a `#[derive(Debug)]` on `Event` and `Behaviour` does not work nicely due to the associated type. We can either require `Debug` on the associated type or manually impl `Debug`, for example as follows:

```suggestion

impl<S: Store> Debug for Event<S>
where
    S::ToSwarm: Debug,
{
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        match self {
            Event::RecordUpdated { peer } => f
                .debug_struct("Event::RecordUpdated")
                .field("peer", peer)
                .finish(),
            Event::Store(to_swarm) => f.debug_tuple("Event::Store").field(to_swarm).finish(),
        }
    }
}
```

We also need to derive `Debug` for some types in `memory_store` for it to be usable like this.